Automattic/wp-calypso
# Stepper: Track flow starting point

Closes [#93061](https://github.com/Automattic/wp-calypso/issues/93061)


## Proposed Changes

* Add a new event, `calypso_stepper_flow_start`, to track flow starts, regardless of whether it is a signup flow or not.
* Add a mechanism to prevent tracking the same event within an interval of 20 minutes.
* Detects when a flow redirects the user to another flow and triggers a new event.


## Testing Instructions

Scenario 1: Event Tracking
* Start any stepper flow (/setup/migration-signup)
* Check if the `calypso_stepper_flow_start` was triggered

Scenario 2: Event deduplication. 
* Start any stepper flow (/setup/migration-signup)
* Check if the `calypso_stepper_flow_start` was triggered
* Reload the page and check if the event is not triggered again

Scenario 3: Same flow different siteId
* Start any stepper flow that requires a site (e.g /setup/site-setup/goal siteSlug={siteSlug}
* Check if the `calypso_stepper_flow_start` was triggered
* Reload the page using a different siteSlug
* Check if a new `calypso_stepper_flow_start` was triggered


Scenario 4: Detect Flow changes
* Got `/start`
* Select any plan
* When you reach the page `What are your goals?` check if the `calypso_stepper_flow_start` was triggered
* Select `Import existing content or website` and continue
* Check if a new event was triggered with the `flow=site-migration` and the `ref=goal`


## Pre-merge Checklist
